test/stages/__init__.py

```diff
@@ -74,29 +71,22 @@ def mock_multicohort() -> MultiCohort:
     # Create a cohort with two datasets
     cohort_a = mc.create_cohort('CohortA')
     # Create a dataset in the cohort (legacy)
-    ds = cohort_a.create_dataset('projecta')
-    # Create a dataset in the multicohort (new)
-    dm1 = mc.add_dataset(ds)
-    # Add sequencing groups to the cohort.dataset AND multicohort.dataset
-    sg1 = add_sg(ds, 'CPGXXXX', external_id='SAMPLE1')
-    sg2 = add_sg(ds, 'CPGAAAA', external_id='SAMPLE2')
-    dm1.add_sequencing_group_object(sg1)
-    dm1.add_sequencing_group_object(sg2)
-
-    ds2 = cohort_a.create_dataset('projectc')
-    dm2 = mc.add_dataset(ds2)
-    sg3 = add_sg(ds2, 'CPGCCCC', external_id='SAMPLE3')
-    sg4 = add_sg(ds2, 'CPGDDDD', external_id='SAMPLE4')
-    dm2.add_sequencing_group_object(sg3)
-    dm2.add_sequencing_group_object(sg4)
+    ds = mc.create_dataset('projecta')
+
+    # Add sequencing groups to the cohort AND dataset
+    cohort_a.add_sequencing_group_object(add_sg(ds, 'CPGXXXX', external_id='SAMPLE1'))
```

Comment:
This is an inner call to `add_sgs` to create the SG in the Dataset, which returns the SG. That SG is fed directly into `Cohort. add_sequencing_group_object`. Saves an extra line.